Sources/Xcodeproj/generate().swift

```diff
@@ -140,23 +140,36 @@ func open(_ path: AbsolutePath, body: ((String) -> Void) throws -> Void) throws
 }
 
 /// Finds directories that will be added as blue folder
-/// Excludes hidden directories and Xcode projects and directories that contains source code
+/// Excludes hidden directories, Xcode projects and reserved directories
 func findDirectoryReferences(path: AbsolutePath) throws -> [AbsolutePath] {
     let rootDirectories = try walk(path, recursively: false)
-    let rootDirectoriesToConsider = rootDirectories.filter {
+
+    return rootDirectories.filter {
         if $0.suffix == ".xcodeproj" { return false }
         if $0.suffix == ".playground" { return false }
         if $0.basename.hasPrefix(".") { return false }
+        if isReservedDirectory($0.basename) { return false }
         return isDirectory($0)
     }
-    
-    let filteredDirectories = try rootDirectoriesToConsider.filter {
-        let directoriesWithSources = try walk($0).filter {
-            guard let fileExt = $0.extension else { return false }
-            return SupportedLanguageExtension.validExtensions.contains(fileExt)
-        }
-        return directoriesWithSources.isEmpty
+}
+
+func isReservedDirectory(_ directory: String) -> Bool {
```

Comment:
For now, I think it would be ok for PackageBuild to expose a function explicitly for this purpose, like `isReservedRootDirectory(name: String)`.

In the longer term I think we need to refactor how this all works, but let's not make this PR more complicated than needs be.